0002-crypto-conditions/0002-crypto-conditions.md

```diff
@@ -114,312 +114,288 @@ This specification is a part of the [Interledger Protocol](https://interledger.o
 
-The SHA-256 feature suite provides the SHA-256 hash function. SHA-256 is a cryptographic hash function published by the US National Institute of Standards and Technology that produces 256 bits of output. This feature suite is assigned the feature bit 2^0 = 0x01.
+The field is encoded as a variable length octet string and specifies the set of types an implementation must support in order to be able to successfully validate the fulfillment of this condition. This is the set of types and subtypes of the top-level condition and all sub-crypto-conditions, recursively. 
 
-## PREIMAGE {#preimage-feature-suite}
-The PREIMAGE feature suite provides conditions that use a preimage as a one-time signature. This feature suite is assigned the feature bit 2^1 = 0x02.
+Each bit represents a type. The list of known types is the IANA-maintained [Crypto-Condition Type Registry](#crypto-conditions-type-registry) .
 
-The fingerprint of a preimage condition is the hash of an arbitrary value. The payload of a preimage fulfillment is the hashed arbitrary value before hashing, also known as the preimage. Conditions that use this preimage MUST also rely on a cryptographically secure hashing algorithm. Since cryptographically secure hashing functions are preimage-resistant, only the original creator of a preimage condition can produce the preimage, as long as it contains a large amount of random entropy.
+The presence of one of more sub-crypto-conditions of a specific type is indicated by setting the numbered bit corresponding to the type number of that type. Bits are numbered from the least significant bit of the first byte to the most significant bit of the last byte.
 
-## PREFIX {#prefix-feature-suite}
-The PREFIX feature suite provides conditions that prepend a fixed message to a subcondition. This feature suite is assigned the feature bit 2^2 = 0x04.
+#### Examples
 
-A prefix condition prepends the message to be validated with a constant string before passing it on to the subcondition's validation function. <!-- TODO: better explanation of why -->
+The following bitmask indicates the presence of the subtypes PREIMAGE-SHA-256 (type number 0) and RSA-SHA-256 (type number 3)
 
-## THRESHOLD {#threshold-feature-suite}
-The THRESHOLD feature suite provides conditions that have several weighted subconditions and a threshold number. This feature suite is assigned the feature bit 2^3 = 0x08.
+    Hex
+    1    2    3    4    
+    0x09 0x00 0x00 0x00
+    
+    Binary
+    1         2         3         4          
+    0000 1001 0000 0000 0000 0000 0000 0000  
+    
+The following bitmask indicates the presence of the subtype ED25519 (type number 4)
 
-Threshold conditions provide flexible multi-signing, such as requiring "M-of-N" subconditions be fulfilled. Subconditions can also be weighted so that one subcondition can count multiple times towards meeting the threshold.
+    Hex
+    1    2    3    4    
+    0x10 0x00 0x00 0x00 
```

Comment:
Those examples don't include the encoding stuff, just the raw bytes that will be encoded. That said they need to be updated because I changed that field to variable length.

While we have defined OER as our canonical encoding format I was trying to avoid being too tightly bound to that and so avoiding putting things like the length indicator in the example.

If the consensus is that we should put it in I'm happy to do so.